### PR TITLE
:sparkles: Signin/Signup from shared link

### PR DESCRIPTION
**Pull request by pabloalba**
Related to https://tree.taiga.io/project/penpot/us/3472

### CODE REVIEW

File: frontend/src/app/main/ui/auth/recovery_request.cljs [hunk 1]
```diff
@@ -22,15 +22,21 @@
...
-  []
+  [{:keys [on-success-callback] :as props}]
   (let [form      (fm/use-form :spec ::recovery-request-form :initial {})
         submitted (mf/use-state false)
 
+        default-success-finish #(st/emit! (dm/info (tr "auth.notifications.recovery-token-sent")))
+
+        success-finish #(if (nil? on-success-callback)
+                   (default-success-finish)
+                   (on-success-callback (:email %)))
```
Comment (niwinz) [inline frontend/src/app/main/ui/auth/recovery_request.cljs]: I think that this anonymos function can be inlined directly in the on-success callback, it is a single use anonymous function of 2 lines.

File: frontend/src/app/main/ui/viewer.cljs [hunk 2]
```diff
@@ -255,6 +257,8 @@
     (when (not allowed)
       (st/emit! (dv/go-to-section :interactions)))
 
+    (st/emit! (modal/hide))
```
Comment (niwinz) [inline frontend/src/app/main/ui/viewer.cljs]: Do not emit events on body of event, you want perform side effects, it is better to wrap it in an effect hook:

```clojure
(mf/with-effect []
  (st/emit! (modal/hide)))
```

Comment (niwinz) [reply frontend/src/app/main/ui/viewer.cljs]: The same for the previous code that is not part of this PR.

File: frontend/src/app/main/ui/viewer/login.cljs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        success-email-sent
+        (fn [email]
+          (reset! user-email email)
+          (set-current-section :email-sent))
+        success-register
+        (fn [data]
+          (reset! register-token (:token data))
+          (set-current-section :register-validate))
+        ]
+    (swap! storage assoc :redirect-url uri)
```
Comment (niwinz) [inline frontend/src/app/main/ui/viewer/login.cljs]: The same, no side effects on render, wrap it with effect hook

**Comment by superalex** [comment]
Great!,

Some stuff I found:

- The header need some space right and bottom:

image

- After login this button doesn't change and keeps saying me to login (if I refresh it shows I'm logged)

**Pull request merged**